pipelines/pipelines.go

```diff
@@ -238,3 +238,19 @@ func ForkMapCtx[S, T any](ctx context.Context, in <-chan S, f func(context.Conte
 	}()
 	return tChan
 }
+
+// Reduce runs a reducer function on every input received from the in chan and returns the output
+func Reduce[S, T string](ctx context.Context, in <-chan S, f func(T, S) T) T {
+	var result T
+	for {
+		select {
+		case <-ctx.Done():
+			return result
+		case s, ok := <-in:
+			if !ok {
+				return result
+			}
+			result = f(result, s)
+		}
+	}
+}
```

Comment:
Would you mind adding additional unit tests to exercise this functionality like the other combinators?

Reply:
yes! added a couple tests